app/lib/pre_assembly/batch.rb

```diff
@@ -90,25 +62,71 @@ def digital_objects
     end
     # rubocop:enable Metrics/AbcSize
 
-    # For each of the passed DigitalObject's ObjectFiles, sets the checksum attribute.
-    # @param [DigitalObject] dobj
-    def load_checksums(dobj)
-      log '  - load_checksums()'
-      dobj.object_files.each { |file| file.provider_md5 = file.md5 }
+    # used by discovery report
+    # @return [Array<DigitalObject>]
+    def objects_to_process
+      @objects_to_process ||= digital_objects.reject { |dobj| skippables&.key?(dobj.container) }
+    end
+
+    private
+
+    def stager
+      staging_style_symlink ? LinkStager : CopyStager
+    end
+
+    # object containers that should be skipped
+    def skippables
+      @skippables ||= begin
+        skippables = {}
+
+        if File.readable?(progress_log_file)
+          docs = YAML.load_stream(File.read(progress_log_file))
+          docs = docs.documents if docs.respond_to? :documents
+          docs.each do |yd|
+            skippables[yd[:container]] = true if yd[:pre_assem_finished]
+          end
+          skippables
+        end
+      end
+    end
+
+    # Discover object containers from the object manifest file suppled in the bundle_dir.
+    def discover_containers_via_manifest
+      manifest_rows.each_with_index do |r, i|
+        next if r[:object]
+        raise 'Missing header row in manifest.csv' if i == 0
+
+        raise "Missing 'object' in row #{i}: #{r}"
+      end
+      manifest_rows.map { |r| bundle_dir_with_path r[:object] }
+    end
+
+    # A method to discover stageable items (i.e. files) with a given object folder.
+    # Takes a root path of the object folder (as supplied in the object manifest).
+    # It then finds all files within with an eager glob pattern.
+    def discover_items_via_crawl(root)
+      Dir.glob("#{root}/**/*")
+    end
+
+    # A convenience method to return all ObjectFiles for all digital objects.
+    # Also used for stubbing during testing.
+    def all_object_files
+      digital_objects.map(&:object_files).flatten
```

Comment:
git thinks this code was added, but it got confused by the methods I moved to private.  I didn't touch this code.